core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -707,6 +708,8 @@ public void copyLogSegmentsToRemote(UnifiedLog log) throws InterruptedException
                 this.cancel();
             } catch (InterruptedException | RetriableException ex) {
                 throw ex;
+            } catch (CorruptIndexException ex) {
+                logger.error("Error occurred while copying log segments. Index appeared to be corrupted for partition: {}  ", topicIdPartition, ex);
```

Comment:
I am assuming that the way this error will be monitored is by creating an alarm on `RemoteCopyLagSegments` [1]. Is that right?

Can you also please why shouldn't we increment the `failedRemoteCopyRequestRate` and `failedRemoteCopyRequestRate` metric that are being incremented in catch exception below?



[1] https://kafka.apache.org/documentation.html#tiered_storage_monitoring

Reply:
Agree